src/Bridges/Constraint/bridges/fix_parametric_variables.jl

```diff
@@ -0,0 +1,161 @@
+    f::MOI.ScalarQuadraticFunction{T}
+    values::Dict{MOI.VariableIndex,Union{Nothing,T}}
+    new_coefs::Dict{MOI.VariableIndex,T}
+end
+
+const FixParametricVariables{T,OT<:MOI.ModelLike} =
+    SingleBridgeOptimizer{FixParametricVariablesBridge{T},OT}
+
+function bridge_constraint(
+    ::Type{FixParametricVariablesBridge{T,S}},
+    model::MOI.ModelLike,
+    f::MOI.ScalarQuadraticFunction{T},
+    s::S,
+) where {T,S<:MOI.AbstractScalarSet}
+    affine = MOI.ScalarAffineFunction(f.affine_terms, f.constant)
+    ci = MOI.add_constraint(model, affine, s)
+    values = Dict{MOI.VariableIndex,Union{Nothing,T}}()
+    new_coefs = Dict{MOI.VariableIndex,T}()
+    for term in f.quadratic_terms
+        values[term.variable_1] = nothing
+        values[term.variable_2] = nothing
+        new_coefs[term.variable_1] = zero(T)
+        new_coefs[term.variable_2] = zero(T)
+    end
+    return FixParametricVariablesBridge{T,S}(ci, f, values, new_coefs)
+end
+
+function MOI.supports_constraint(
+    ::Type{<:FixParametricVariablesBridge{T}},
+    ::Type{MOI.ScalarQuadraticFunction{T}},
+    ::Type{<:MOI.AbstractScalarSet},
+) where {T}
+    return true
+end
+
+function concrete_bridge_type(
+    ::Type{<:FixParametricVariablesBridge},
+    ::Type{MOI.ScalarQuadraticFunction{T}},
+    ::Type{S},
+) where {T,S<:MOI.AbstractScalarSet}
+    return FixParametricVariablesBridge{T,S}
+end
+
+function MOI.Bridges.added_constrained_variable_types(
+    ::Type{<:FixParametricVariablesBridge},
+)
+    return Tuple{Type}[]
+end
+
+function MOI.Bridges.added_constraint_types(
+    ::Type{FixParametricVariablesBridge{T,S}},
+) where {T,S}
+    return Tuple{Type,Type}[(MOI.ScalarAffineFunction{T}, S)]
+end
+
+function MOI.get(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintFunction,
+    bridge::FixParametricVariablesBridge,
+)
+    return bridge.f
+end
+
+function MOI.get(
+    model::MOI.ModelLike,
+    ::MOI.ConstraintSet,
+    bridge::FixParametricVariablesBridge,
+)
+    return MOI.get(model, MOI.ConstraintSet(), bridge.affine_constraint)
+end
+
+function MOI.delete(model::MOI.ModelLike, bridge::FixParametricVariablesBridge)
+    MOI.delete(model, bridge.affine_constraint)
+    return
+end
+
+MOI.get(::FixParametricVariablesBridge, ::MOI.NumberOfVariables)::Int64 = 0
+
+function MOI.get(::FixParametricVariablesBridge, ::MOI.ListOfVariableIndices)
+    return MOI.VariableIndex[]
+end
+
+function MOI.get(
+    bridge::FixParametricVariablesBridge{T,S},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T},S},
+)::Int64 where {T,S}
+    return 1
+end
+
+function MOI.get(
+    bridge::FixParametricVariablesBridge{T,S},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T},S},
+) where {T,S}
+    return [bridge.affine_constraint]
+end
+
+function MOI.modify(
+    model::MOI.ModelLike,
+    bridge::FixParametricVariablesBridge{T,S},
+    chg::MOI.ScalarCoefficientChange{T},
+) where {T,S}
+    MOI.modify(model, bridge.affine_constraint, chg)
```

Comment:
I don't this that this would work, the coefficient change should change the coefficient before we added the quadratic part. If `final_touch` was already called then we should add the coefficient here

Reply:
Won't final touch need to be called after this anyway?

There are two parts:

 * We modify the existing `.affine_constraint`. This catches adding new variables, or modifying a variable that doesn't exist in the quadratic part
 * We modify `bridge.f`, so that the next time we call `final_touch`, it gets updated correctly.